abci/precommit/precommit.go

```diff
@@ -0,0 +1,96 @@
+package precommit
+
+import (
+	"cosmossdk.io/log"
+	"github.com/cosmos/cosmos-sdk/types/module"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/skip-mev/block-sdk/block"
+)
+
+// MempoolEvictionPreCommiter is a PreCommit function that evicts txs that invalid from the app-side mempool.
+type MempoolEvictionPreCommiter struct {
+	// logger
+	logger log.Logger
+
+	// app side mempool interface
+	mempl block.Mempool
+
```

Comment:
I'd add a sub PreCommit hook here so we can wrap existing functionality w/ this one, i.e Polaris has their own Precommit and wants to extend to have Mempool Eviction